api/v1alpha1/dnspolicy_types.go

```diff
@@ -0,0 +1,346 @@
+	//
+	// Route53: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-policy-weighted.html
+	// +kubebuilder:default=120
+	DefaultWeight Weight `json:"defaultWeight,omitempty"`
+	// +optional
+	Custom []*CustomWeight `json:"custom,omitempty"`
+}
+
+type GeoCode string
+
+func (gc GeoCode) IsDefaultCode() bool {
+	return gc == DefaultGeo
+}
+
+func (gc GeoCode) IsWildcard() bool {
+	return gc == WildcardGeo
+}
+
+type LoadBalancingGeo struct {
+	// defaultGeo is the country/continent/region code to use when no other can be determined for a dns target cluster.
+	//
+	// The values accepted are determined by the target dns provider, please refer to the appropriate docs below.
+	//
+	// Route53: https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-values-geo.html
+	// +required
+	DefaultGeo string `json:"defaultGeo,omitempty"`
+}
+
+// DNSPolicyStatus defines the observed state of DNSPolicy
+type DNSPolicyStatus struct {
+
+	// conditions are any conditions associated with the policy
+	//
+	// If configuring the policy fails, the "Failed" condition will be set with a
+	// reason and message describing the cause of the failure.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// observedGeneration is the most recently observed generation of the
+	// DNSPolicy.  When the DNSPolicy is updated, the controller updates the
+	// corresponding configuration. If an update fails, that failure is
+	// recorded in the status condition
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	HealthCheck *HealthCheckStatus `json:"healthCheck,omitempty"`
+}
+
+var _ common.KuadrantPolicy = &DNSPolicy{}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="gateway.networking.k8s.io/policy=direct"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[0].reason`,description="DNSPolicy Status",priority=2
+// +kubebuilder:printcolumn:name="TargetRefKind",type="string",JSONPath=".spec.targetRef.kind",description="Type of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="TargetRefName",type="string",JSONPath=".spec.targetRef.name",description="Name of the referenced Gateway API resource",priority=2
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+// DNSPolicy is the Schema for the dnspolicies API
+type DNSPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DNSPolicySpec   `json:"spec,omitempty"`
+	Status DNSPolicyStatus `json:"status,omitempty"`
+}
+
+func (p *DNSPolicy) GetWrappedNamespace() gatewayapiv1.Namespace {
+	return gatewayapiv1.Namespace(p.Namespace)
+}
+
+func (p *DNSPolicy) GetRulesHostnames() []string {
+	//TODO implement me
+	panic("implement me")
```

Comment:
guessing this was already here. Would it be better to return an empty slice or do we want to keep the panic

Reply:
fixed

Reply:
yeah probably not, we can have return an empty array